data_structures/src/chain.rs

```diff
@@ -1052,6 +1052,21 @@ impl DataRequestOutput {
             .and_then(|res| res.checked_add(self.tally_fee))
             .ok_or_else(|| TransactionError::FeeOverflow)
     }
+
+    /// Returns the DataRequestOutput weight
+    pub fn weight(&self) -> u32 {
+        // Witness reward: 8 bytes
+        // Witnesses: 2 bytes
+        // commit_fee: 8 bytes
+        // reveal_fee: 8 bytes
+        // tally_fee: 8 bytes
+        // min_consensus_percentage: 4 bytes
+        // collateral: 8 bytes
+
+        self.data_request
+            .weight()
+            .saturating_add(8 + 2 + 8 + 8 + 8 + 4 + 8)
+    }
```

Comment:
I see that we are renouncing to the chance to add the variable part here.

Simplest implementation could look like this:
```
reducers = 2 bytes (1 each)
filters = sum of the CBOR bytes lengths
retrieval scripts = sum of the CBOR bytes lengths
```